tests/test_utils.py

```diff
@@ -172,3 +174,11 @@ def test_p_values_from_sample():
 
     ks_results = insertion_p_value(ns.insertion[nlive:-nlive], nlive, batch=1)
     assert ks_results['p-value'] > 0.05
+
+
+@pytest.mark.parametrize('method', ['channel', 'kish'])
+def test_effective_samples(method):
+    w = np.ones(20)
+    assert effective_samples(w, method=method) == approx(20, rel=1e-6)
+    w[10:] = 1e-3
+    assert effective_samples(w, method=method) == approx(10, rel=1e-2)
```

Comment:
Note, `gamma=0.5` leads to higher `n_eff` and will require either further reducing the `1e-3` and/or relaxing the relative error.